### ISSUE TITLE
Allow "present" for interviewee bio job end dates.

### ISSUE THREAD

**Issue by rachellane12** [sciencehistory/scihist_digicoll#2045]
It would be great to have an empty end date for the interviewee bio. Sometimes interviewees are still in a position and right now, there's no way to not have a date. Also, writing "present" doesn't work because it only accepts dates. Example: It's not clear from Helen Donis-Keller's [interviewee bio](https://digital.sciencehistory.org/works/uwilrto) that she is still working in her positions at Olin College of Engineering. Dr. Shirley Malcom is another example. She's still in her position at AAAS.

## Later note from @rachellane12 :
 I've left blank end dates before, but then it showed that Donis-Keller for example was a professor at Olin only in 2001, which obviously isn't accurate since she's still there. I'm open to whatever solution is best to address that situation, but I do think it would be beneficial to have some way to indicate that someone is currently in a position.

**Comment by eddierubeiz** [comment]
@rachellane12: I've taken a crack at this. Take a look at 
- https://staging-digital.sciencehistory.org/works/uwilrto
- https://staging-digital.sciencehistory.org/admin/interviewee_biographies/6/edit when you have a minute please (presumably on Tuesday :) ).

If we go with this solution, you can (and should) enter the word "present" in the date field if you want the job to read "to present" on the front end, as opposed to leaving the end date blank. (If you leave it blank, the front end will just show the start date, as before.

**Comment by rachellane12** [comment]
@eddierubeiz This looks great. Exactly what I had in mind for interviewees who happen to still be working.

**Comment by rachellane12** [comment]
@eddierubeiz That seems like a good solution because it addresses both needs that we have. Thanks for looking into this!

**Comment by eddierubeiz** [comment]
Sounds good!
@rachellane12  - just to be consistent, let's go over a dozen or so interviewees that currently have blank job end dates in their bios.
(Just fill in an end date identical to the start date unless you really want the job to read "to present".)
This is probably easier to do after the new code is in production, but whatever works for you!

|Interviewee |URL | Jobs with blank endings|
|----|----|----|
Tsukiyama | https://digital.sciencehistory.org/admin/interviewee_biographies/5820/edit |  Postodoctoral Fellow in the laboratory of Carl Wu; Associate Member; Assistant Professor
Sha| https://digital.sciencehistory.org/admin/interviewee_biographies/5785/edit |  Research; Postdoctoral work; Postdoctoral work;
Gall | https://digital.sciencehistory.org/admin/interviewee_biographies/5562/edit |  ( this one has several wonky jobs; please investigate)
Lee| https://digital.sciencehistory.org/admin/interviewee_biographies/5666/edit |  Postdoctoral Fellow; Professor, Clayton Foundation Laboratories for Peptide Biology
Waldor | https://digital.sciencehistory.org/admin/interviewee_biographies/5828/edit |  Residency; Clinical Fellowship in Infectious Diseases; Postdoctoral Fellowship with John J. Mekalanos
Zheng| https://digital.sciencehistory.org/admin/interviewee_biographies/5856/edit |  HHMI Investigator
Sadler| https://digital.sciencehistory.org/admin/interviewee_biographies/5772/edit |  Chairman
Luskin  | https://digital.sciencehistory.org/admin/interviewee_biographies/5917/edit |  Assistant Professor, Anatomy and Cell Biology
Newman | https://digital.sciencehistory.org/admin/interviewee_biographies/5725/edit |  Director, NIDA-IRP Medication Development Program; Chief, Molecular Targets and Medications Discovery Branch Chief, Medicinal Chemistry Section; Scientific Director, National Institute on Drug Abuse Intramural Research Program
Enke | https://digital.sciencehistory.org/admin/interviewee_biographies/1/edit |  Professor Emeritus
Solnica-Krezel | https://digital.sciencehistory.org/admin/interviewee_biographies/7517/edit |  Professor and Head, Department of Developmental Biology
Diedrich | https://digital.sciencehistory.org/admin/interviewee_biographies/5/edit |  Infectious Disease Research Staff Scientist; Founder and President
Panning | https://digital.sciencehistory.org/admin/interviewee_biographies/5939/edit |  AssociateProfessor, Biochemistry & Biophysics
Anagnostopoulos | https://digital.sciencehistory.org/admin/interviewee_biographies/9/edit |  Research Chemist; Monsanto Fellow; Corporate Officer and Vice Chairman of Corporate Development and Growth
Augustine | https://digital.sciencehistory.org/admin/interviewee_biographies/10/edit |  Acting Secretary of the Army; Chairman; Chairman; President and Chairman; Founder; Chairman; Chairman; Member; Member
 Leone | https://digital.sciencehistory.org/admin/interviewee_biographies/5667/edit |  Director of the Medical College of Wisconsin Cancer Center

**Comment by eddierubeiz** [comment]
@rachellane12 I fixed the first problem, which was due to an oversight on my part - sorry about that! See e.g. https://digital.sciencehistory.org/works/v6kmkyc .

**Comment by rachellane12** [comment]
@eddierubeiz Wonderful! Thanks for doing that.

@jrochkind Regarding the second issue, yes, I do mean the dates of jobs listed in the biographical material. Before how things were handled is that only the start date was listed if no end date was listed, but this made it seem that the interviewee was only in that position for a year (because we do have examples of people being in positions for just a year). I originally noticed this when we had interviewees like Donis-Keller and Malcom who were at their current positions and weren't just there for one year and I wanted it to show correctly, hence my thought of "present."

Obviously, there is a lot of stuff that has come up since my initial discovery, namely, that there are interviews that don't have any job start/end dates at all (I didn't realize that) and at least one interview where I wasn't able to find an end date for a position online (ex: Waldor). I honestly didn't expect as many interviewees with blank job end dates as Eddie identified.

I do think it could be good to have a broader discussion about how to handle updating metadata in the DC specifically regarding "present." As of right now, we update job information whenever an interviewee lets us know about a change in position or if it's something that we notice.

I don't exactly know what the solution to all of this is, but I do think that there should be a way to distinguish when an interviewee holds a position for just one year and when all we know is the start date. I'm all for talking on Zoom next week. I want to give you the information you need so that we can come up with the best solution.

**Comment by eddierubeiz** [comment]
For the second problem (you have a start date; you know the person isn't in the position any more; you don't know when the person stopped working there): granted, there's some ambiguity about how to interpret the absence of an end date. In other words, the table above should really be:

In real life | Start field in the form | End field in the form | Displayed to users
-- | -- | -- | --
Considered to be still working there | 2012 | blank | 2012 to present
Only worked there for a year | 2012 | 2012 | 2012
Started there in 2012; end date unknown | 2012 | 2012 | 2012

**Comment by jrochkind** [comment]
Is the second category still right now displaying the same way it did before we started changing things? 

I think yeah we need to take a big picture look at these dates and develop a "model" based on considering them in whole, not just the latest "problem" we mentioned. 

We can also consider how high a priority this really is -- if it's been displaying this way for 6-7 years with nobody noticing/complaining until now, perhaps it's not super high priority to fix? But maybe it is!  But we have infinite things we could make better in the Digital Collections, we ("we" being all of us, stakeholders) need to consider what improvements are worth how much time when.

**Comment by rachellane12** [comment]
@eddierubeiz Yes, that table is accurate. I appreciate you putting it in that form because it helps make our discussion more clear.

@jrochkind Regarding the larger issue, I agree that a discussion will help to avoid going in circles and to identify the priority. I will say that I have had interviewees reach out to correct their information and ask that it be updated and in one case, it was a user who had identified an error in chronological information about an interviewee (I will see if I can find that example). Honestly, it would be so much easier if interviewees kept in touch because then it's an easily solved problem!

**Comment by apinkney0696** [comment]
Hi y'all, just popping in to offer my two cents. 

It's not our business how OH chooses to update biographical metadata as new information comes about, but I do suggest establishing a workflow to verify the data is correct every so often. Maybe this is when an interviewee dies, maybe you'll go through living interviewees' data annually, or maybe you just want to make updates when someone reaches out to correct the data - it's up to you, but it's likely a decision you'll want to have documented somewhere. If you'd like help with making this decision, I'd be happy to help advise you as I'm sure any of the other cataloging folks would be!

It sounds like we have fixed the issue of providing a "present" option for interviewee bio dates. I'm going to close this ticket as the original task has been completed. 

Now a new need has presented itself: the ability to have "unknown" dates within user bios. At this point, I suggest going through the collection to identify if there are any additional interviewee bios that don't fit into the following categories we've already established:

- dates are known
- dates are known, currently holding position (will need to be updated eventually)
- one or both dates are unknown (may need to be updated if additional information is uncovered)

Try to find out if there are any other situations where the date metadata you have isn't working with our input form, and try to determine what changes will be most intuitive for you and the end-user in each situation. We may have a few weird outliers, and that's okay! Once you've done that, please make another ticket to address the unknown date issue and other situations as needed, then we'll go from there!

Does this sound like a good plan to everyone? Thanks!

**Comment by rachellane12** [comment]
@apinkney0696 That all makes sense to me. I will check in with Dave about this plan and after doing the research you've suggested create a new ticket for "unknown" dates.
